### PR TITLE
Replace InMemoryAssetResolver with DynamicTextureProvider

### PR DESCRIPTION
**Pull request by lilleyse**
Fixes https://github.com/CesiumGS/cesium-omniverse/issues/116

Removes `InMemoryAssetResolver` and uses Omniverse's built-in `DynamicTextureProvider` instead. This has several benefits:

* No longer need to convert raw pixels to BMP
* Textures now work when Cesium for Omniverse is first installed, without a restart [^1]
* Compressed textures and mipmaps are supported in `DynamicTextureProvider`. That can be a follow up PR.
* Pixel data can be swapped out dynamically. This means we don't need to create a new material when higher detail imagery comes in. Going a step further, this allows us to do basic material pooling which could significantly improve performance. I haven't actually tested this yet, it will be a follow up PR.
* I no longer see cyan flashes (texture loading) for 3D Tiles input. Red flashes still happen, but that's expected.
* Removed `/rtx-transient/resourcemanager/enableTextureStreaming = false` workaround from https://github.com/CesiumGS/cesium-omniverse/issues/95

This doesn't fix the crashes / hangs we've noticed. At least not all of them. I can still trigger hangs by reloading agi.usda repeatedly.

One potential blocker is that each time a texture is loaded a warning gets printed. I reached out to the Nvidia team about this.

```
2023-03-29 19:50:32  [Warning] [omni.hydra] Material parameter 'tex' was assigned an incompatible texture: 'dynamic://imagery_ion_2__11718102_184221417_4827478_456288989__11717949_310164846_4827677_90915967'
```

image

To do:

* [x] Fix Windows CI

[^1]: Though a restart is still required to register Cesium USD schemas

### CODE REVIEW

File: src/core/src/DynamicTextureProviderCache.cpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    bool removeItem = false;
+
+    {
+        // Make sure the accessor goes out of scope and the item is unlocked before removing the item
+        DynamicTextureProviderCache::Accessor accessor;
+        auto found = _items.find(accessor, name);
+        if (found) {
+            auto& referenceCount = accessor->second->_referenceCount;
+            referenceCount--;
+            if (referenceCount == 0) {
```
Comment (weegeekps) [inline src/core/src/DynamicTextureProviderCache.cpp:29]: As long as `found` is held it blocks other threads, correct? I'm concerned about if there's a case where we could run into a situation where reference count could go negative.

**Review by weegeekps — COMMENTED** [review]
Looks good. One open question.

File: src/core/src/DynamicTextureProviderCache.cpp (hunk 1 above)
Comment (lilleyse) [reply src/core/src/DynamicTextureProviderCache.cpp:29]: I believe `accessor` acts as a lock guard, so we shouldn't see any race conditions

Comment (weegeekps) [reply src/core/src/DynamicTextureProviderCache.cpp:29]: Sounds good. This is good to go once you fix the Windows builds. I'll give it a quick once over when that's done.

**Review by weegeekps — APPROVED** [review]
:rocket:

**Comment by lilleyse** [comment]
This PR doesn't seem to fix the blue dots artifact that shows up in Create when a material is loaded. Interestingly this doesn't happen in `cesium.omniverse.app.kit`.

blue-flickers

This is different from the cyan placeholder when loading textures, which is fixed in this PR.

**Review by weegeekps — APPROVED** [review]
Good to go.

**Pull request merged**